src/main/java/software/amazon/awssdk/crt/http/CrtHttpStreamHandler.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.crt.http;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Interface that Native code knows how to call when handling Http Request/Responses
+ *
+ * Maps 1-1 to the Native Http API here: https://github.com/awslabs/aws-c-http/blob/master/include/aws/http/request_response.h
+ */
+public interface CrtHttpStreamHandler {
+
+    /**
+     * Called from Native when new Http Headers have been received.
+     * Note that this function may be called multiple times as HTTP headers are received.
+     *
+     * @param stream The HttpStream object
+     * @param responseStatusCode The HTTP Response Status Code
+     * @param nextHeaders The headers received in the latest IO event.
+     */
+    void onResponseHeaders(HttpStream stream, int responseStatusCode, HttpHeader[] nextHeaders);
+
+    /**
+     * Called from Native once all HTTP Headers are processed. Will not be called if there are no Http Headers in the
+     * response. Guaranteed to be called exactly once if there is at least 1 Header.
+     *
+     * @param stream The HttpStream object
+     * @param hasBody True if the HTTP Response had a Body, false otherwise.
+     */
+    void onResponseHeadersDone(HttpStream stream, boolean hasBody);
```

Comment:
This should be defaulted

Reply:
I wanted this PR to be more focused on the Java interface than the implementation. I'd prefer to punt the Java implementation to a later PR, or have the implementation live in [aws-sdk-java-v2](https://github.com/aws/aws-sdk-java-v2/tree/master/http-clients).

Reply:
Yes, that's why I'm bringing it up. Java interfaces can have a [default implementation](https://docs.oracle.com/javase/tutorial/java/IandI/defaultmethods.html) (which in all of the cases I've highlighted should do nothing). The important thing here is that an implementer is only required to implement onHeader and onComplete.